DP-29104 eliminate lightest font weight in the system

clairesunstudio opened the pull request:
Changed:
- Removed the lightest font-weight for text in the system
- Space out font-weight scale evenly to better show visual hierarchy, bump up all text font weights by at least 50
- Consolidate hard-coded font weights into variable
- Remove styling overrides on telephone links, so they appear as other contact links 

Before:
```scss
$fonts-lightest: 150 !default; // only used for styling non text font weights, e.g. separator |
$fonts-lighter:  250 !default; (removed)
$fonts-light:    350 !default; // body font weight
$fonts-normal:   400 !default; // (bumped up to 450)
$fonts-bold:     550 !default; 
$fonts-bolder:   650 !default;
```

After:
```scss
$fonts-lightest: 150 !default; // only used for styling non text font weights, e.g. separator |
$fonts-light:    400 !default; // body font weight
$fonts-normal:   500 !default;
$fonts-semibold: 550 !default;
$fonts-bold:     600 !default;
$fonts-bolder:   700 !default;
```


Before:
![Screen Shot 2023-08-24 at 11 01 50 AM](https://github.com/massgov/mayflower/assets/5789411/a43fabe0-d7fa-44ae-bb03-ca9b9247cf9a)


After:
![Screen Shot 2023-08-24 at 4 38 47 PM](https://github.com/massgov/mayflower/assets/5789411/f5f0f0d1-09b4-4e98-8a77-b3a79d2c8fae)

clairesunstudio committed:
eliminate lightest font weight

clairesunstudio committed:
space out font weights

clairesunstudio committed:
convert font weight to use defined vars

clairesunstudio committed:
convert to var and bump up normal

clairesunstudio committed:
add semi bold font weight to vars

clairesunstudio committed:
remove phone override styles

clairesunstudio committed:
remove override of phone number hover style

clairesunstudio committed:
remove unused class that causes discrepancy between patternlab views

clairesunstudio committed:
style lint

clairesunstudio committed:
add changelog

ygannett committed:
Merge branch 'develop' into DP-29104-adjust-global-font-weights

ygannett commented:
@clairesunstudio Thank you for your work.

For the review, I used the [table below](https://www.w3.org/WAI/GL/WCAG3/2020/methods/font-characteristics-contrast/) for reference.  This is a part of WCAG 3.0 support resources.  It's still under development, but WCAG 2.1 doesn't have anything solid to evaluate font weights for their appropriation as reasonable readability.  I hope this gives us some sort of baseline to work with.
<img width="736" alt="apca_contrast-lookup-table" src="https://github.com/massgov/mayflower/assets/9633303/e2335c0f-167b-48a6-b105-f75d3acafb78">

I also want to point out that Noto font is generally thinner than the reference fonts W3C uses (Helvetica Neue, Helvetica, Fira Sans, Kanit, and Arial) as you see below:
<img width="667" alt="sample-fontweight" src="https://github.com/massgov/mayflower/assets/9633303/ccc4ab4a-e94d-40b0-b2f5-5a28fde90b4a">
It seems Noto's font-weight is 100 lighter than those reference fonts.  For example, to match the font-weight of 350 of those fonts visually, it seems Noto's font-weight needs to be 450 as I played around.  We probably need to set the base font-weight heavier for text.  This is something to think about when WCAG 3.0 is released around 2026.  I'm open to discussion.

With the current base font-weight of 350, the following components appear to need a heavier weight:
1. ma__contact-group__details
     - patterns/05-pages-bio-page/05-pages-bio-page.html
     <img width="891" alt="bio-oage_ma__contact-group__details" src="https://github.com/massgov/mayflower/assets/9633303/db843fb2-219a-4438-a719-3b6b74fa5338">

     - patterns/05-pages-contact-row-examples/05-pages-contact-row-examples.html
     <img width="1712" alt="contact-row_ma__contact-group__details" src="https://github.com/massgov/mayflower/assets/9633303/5be5345d-2fe4-4c81-9996-dbade9fe943b">

1. ma__figure__caption
     - patterns/05-pages-information-details/05-pages-information-details.html
     <img width="1487" alt="figure-caption" src="https://github.com/massgov/mayflower/assets/9633303/9840a7de-3c26-4403-bbdb-7b847fd4142e">

1. ma__image-credit__container
     - patterns/05-pages-information-details/05-pages-information-details.html
     <img width="1478" alt="image-creadit" src="https://github.com/massgov/mayflower/assets/9633303/d83f7781-8d91-4d25-a607-dcbc7f4744e2">

1. ma__action-step__header span
     - patterns/05-pages-location-park-content/05-pages-location-park-content.html
<img width="1482" alt="alert-updatetimestamp" src="https://github.com/massgov/mayflower/assets/9633303/f248e66c-b617-44b8-af06-43cd02a9d8cd">

1. secondary-label-close span
     - patterns/05-pages-policy-advisory-directive/05-pages-policy-advisory-directive.html
<img width="1022" alt="close-symbol" src="https://github.com/massgov/mayflower/assets/9633303/9e2d20df-0354-4957-9320-4cbd4a8a035a">

clairesunstudio committed:
Merge branch 'develop' into DP-29104-adjust-global-font-weights

clairesunstudio committed:
adjust button close font width and styles to match the others

clairesunstudio commented:
@ygannett thanks for your thorough review - I fixed the font weight for `x` used as the close icon and adjusted to visually match the others in the system. 

For the rest of the problem areas, based on your research, I think it makes sense to systematically bump up the values by 50, so the lightest font weight variable value in Mayflower for text doesn't fail accessibility. Here's what I changed --> https://github.com/massgov/mayflower/pull/1841/files. This makes all the fonts slightly thicker which will benefit accessibility overall and address the problematic components. Not guaranteed to meet 3.0 yet, but a step closer for sure.

ygannett reviewed (approved):
Thank you for the changes.  No need to meet the 3.0 guideline yet.  That is something to keep in mind as an up-coming change.  That's why I wanted to mention that.  Its approach is more practical for users who are affected by readability issues.  So, accommodating the approach would help those users for sure.

The changes look good.  Once the backstop references are updated, ready to merge.

clairesunstudio committed:
bump up all font weights (#1841)

* bump up all font weights

* style linting

clairesunstudio committed:
approve patternlab backstop

clairesunstudio committed:
approve react backstop

clairesunstudio commented on changelogs/DP-29104.yml:
```diff
@@ -0,0 +1,11 @@
+Changed:
+  - project: Assets
+    component: Variables
+    description: Remove the lightest font-weight for text in the system. Space out font-weight scale evenly to better show visual hierarchy.  Consolidate hard-coded font weights into a new variable. (#1831)
```
```suggestion
    description: Remove the lightest font-weight for text in the system. Space out font-weight scale evenly to better show visual hierarchy.  Consolidate hard-coded font weights into a new variable. (#1831)
```
```suggestion
    description: Remove the lightest font-weight for text in the system. Space out font-weight scale evenly to better show visual hierarchy, bump up all text font weights by at least 50.  Consolidate hard-coded font weights into a new variable. (#1831)
```

clairesunstudio committed:
Update changelogs/DP-29104.yml
